src/pip/_internal/req/constructors.py

```diff
@@ -218,7 +218,7 @@ def parse_req_from_editable(editable_req):
 
 def install_req_from_editable(
     editable_req,  # type: str
-    comes_from=None,  # type: Optional[str]
+    comes_from=None,  # type: Optional[Union[InstallRequirement, str]]
```

Comment:
Extreme nit: `InstallRequirement.__init__` spells this as `Optional[Union[str, InstallRequirement]]`. Maybe put the union arguments in the same order here, for consistency?

Less of a nit, more of a note about typing, when I first read this, my thought was "how do we know it's valid to pass an `InstallRequirement` here. It wasn't obvious to me that if it was invalid, that would have been picked up as a typing error in the `return InstallRequirement` statement below... I thought about suggesting a comment here to explain, but on reflection, this is exactly what type annotations are *for*. So I guess, I learned something about mypy today 🙂